hedera-node/test-clients/README.md

```diff
@@ -1,11 +1,250 @@
+execution requires registering Jupiter extensions to manage the test network lifecycle and make it the target
+of each `HapiSpec`.
+
+The default Gradle `test` task instructs these extensions to create the target network by spawning four
+child subprocesses as nodes. Meanwhile, the `testEmbedded` and `testRepeatable` tasks instruct the
+extensions to embed a single `Hedera` instance in the test process, invoking its workflows directly.
+In embedded mode, there is no hashgraph consensus because the `Platform` is replaced with a simple mock,
+and the `HederaState` implementation uses in-memory data structures instead of a Merkle tree.
+
+**Table of contents**
+
+- [Structure of a `HapiSpec`](#structure-of-a-hapispec)
+- [`HederaNetwork` implementations](#hederanetwork-implementations)
+  * [`RemoteNetwork`](#remotenetwork)
+  * [`SubProcessNetwork`](#subprocessnetwork)
+  * [Embedded networks](#embedded-networks)
+- [Style guide](#style-guide)
+  * [The `@HapiTest` checklist](#the-hapitest-checklist)
+- [JUnit Jupiter integrations](#junit-jupiter-integrations)
+  * [Meta-annotations](#meta-annotations)
+  * [`SharedNetworkLauncherSessionListener`](#sharednetworklaunchersessionlistener)
+  * [`NetworkTargetingExtension`](#networktargetingextension)
+  * [`SpecNamingExtension`](#specnamingextension)
+  * [`SpecManagerExtension`](#specmanagerextension)
+  * [`SpecEntityExtension`](#specentityextension)
+
+## Structure of a `HapiSpec`
+
+A `HapiSpec` groups a sequence of `SpecOperation`s and gives them shared infrastructure and context. When
+the spec is executed against its target `HederaNetwork`, it runs each operation in sequence, and terminates
+with an exception if any operation fails. This is consistent with JUnit test semantics, where a test passes
+as long as no exception is thrown.
+
+Most operations are subclasses of `HapiSpecOperation`, which provides implementation support for submitting
+transactions and sending queries to the target network. The below diagram is a schematic of a `HapiSpec`.
+
+<img src="./docs/assets/hapispec-schematic.png" width="900">
+
+Operations share context through their spec's `HapiSpecRegistry`. The registry maps from `String` names to
+arbitrary values; which are themselves usually references to entities on the target network. For example,
+after a `HapiCryptoCreate` operation submits a transaction that creates a new account, it stores the new
+account's id (say, `AccountID[accountNum=1001]`) in the registry with the name given by the test author
+(say, `"Alice"`). The test author can then write a following `HapiCryptoUpdate` operation that references
+the account by its stable name `"Alice"`, and not worry that the next time the spec runs the created account
+may have a different id on the target network.
+
+Besides providing the context of its registry and target network, a `HapiSpec` also supports its operations
+with some useful infrastructure components. The four most important components are,
```

Comment:
Q: Is there a component for setting config override as it relates to feature flags of non `RemoteNetwork` options?

Reply:
Hmm no there isn't; but that's an interesting idea. 🤔